tests/test_document.py

```diff
@@ -0,0 +1,67 @@
+import os
+import tempfile
+
+import pytest
+
+from dangerzone.document import DocumentFilenameException, DocumentHolder
+
+from . import sample_doc, unreadable_pdf, unwriteable_pdf
+
+
+def test_input_sample_init(sample_doc):
+    DocumentHolder(sample_doc)
+
+
+def test_input_sample_after(sample_doc):
+    d = DocumentHolder()
+    d.input_filename = sample_doc
+
+
+def test_input_file_none():
+    """
+    Attempts to read a document's filename when no doc has been set
+    """
+    d = DocumentHolder()
+    with pytest.raises(DocumentFilenameException):
+        d.input_filename
+
+
+def test_input_file_non_existing():
+    with pytest.raises(DocumentFilenameException):
+        DocumentHolder("fake-dir/non-existing-file.pdf")
```

Comment:
I need to fix the `/` and make it platform-independent. This won't work on Windows.

Reply:
Done.